src/internet_identity/src/anchor_management/registration.rs

```diff
@@ -139,10 +160,22 @@ fn create_captcha<T: RngCore>(rng: T) -> (Base64, String) {
 
 // Check whether the CAPTCHA challenge was solved
 fn check_challenge(res: ChallengeAttempt) -> Result<(), ()> {
+    // avoid processing too many characters
+    if res.chars.len() > CAPTCHA_LENGTH {
+        return Err(());
+    }
+    // Normalize challenge attempts by replacing characters that are not in the captcha character set
+    // with the respective replacement from CHAR_REPLACEMENTS.
+    let normalized_challenge: String = res
```

Comment:
```suggestion
    let normalized_challenge_res: String = res
```

got confused and thought you were re-normalizing the challenge